src/test/stateful_modelinstance_test.cpp

```diff
@@ -81,12 +82,37 @@ class StatefulModelInstanceTempDir : public TestWithTempDir {
             std::tuple<ovms::shape_t, tensorflow::DataType>{{1, 10}, tensorflow::DataType::DT_FLOAT}}};
     }
 
+    void TearDown() override {
+        modelInput.clear();
+    }
+};
+
+class StatefulModelInstanceInputValidation : public ::testing::Test {
+public:
+    inputs_info_t modelInput;
+    std::pair<std::string, std::tuple<ovms::shape_t, tensorflow::DataType>> sequenceId;
+    std::pair<std::string, std::tuple<ovms::shape_t, tensorflow::DataType>> sequenceControlStart;
+
+    void SetUp() override {
+        modelInput = {};
+    }
+
     void TearDown() override {
         TestWithTempDir::TearDown();
         modelInput.clear();
     }
 };
 
+class MockedValidateStatefulModelInstance : public ovms::StatefulModelInstance {
+public:
+    MockedValidateStatefulModelInstance(const std::string& name, ovms::model_version_t version) :
+        StatefulModelInstance(name, version) {}
+
+    const ovms::Status mockValidate(const tensorflow::serving::PredictRequest* request, ovms::ProcessingSpec* processingSpecPtr) {
```

Comment:
Can we use SequenceProcessingSpec reference? 
If necessary please change validate method to take SequenceProcessingSpec& instead of ProcessingSpec*